File: AffiliateMarketing.Domain/Entities/ClickEvent.cs
namespace AffiliateMarketing.Domain.Entities;

public sealed class ClickEvent
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string TrackingId { get; set; } = default!;
    public string IpAddress { get; set; } = default!;
    public string? UserAgent { get; set; }
    public string? Platform { get; set; }
    public Guid ProductId { get; set; }
    public DateTime ClickedAt { get; set; } = DateTime.UtcNow;
    //public Guid AffiliateUserId { get; set; }

    // --- High-Resolution Geo Data ---
    public string? CountryCode { get; set; }
    public string? City { get; set; }
    public string? Region { get; set; }
    public string? Location { get; set; } // Formatted string for dashboard display
}

File: affiliate_api/app/models/click_event.py
# Source: AffiliateMarketing.Domain/Entities/ClickEvent.cs
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, DateTime, ForeignKey
from sqlalchemy.orm import mapped_column, Mapped
from app.core.db import Base

class ClickEvent(Base):
    __tablename__ = "click_events"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    tracking_id: Mapped[str] = mapped_column(String, nullable=False)
    ip_address: Mapped[str] = mapped_column(String, nullable=False)
    user_agent: Mapped[str | None] = mapped_column(String, nullable=True)
    platform: Mapped[str | None] = mapped_column(String, nullable=True)
    product_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("products.id"), nullable=False)
    clicked_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    country_code: Mapped[str | None] = mapped_column(String, nullable=True)
    city: Mapped[str | None] = mapped_column(String, nullable=True)
    region: Mapped[str | None] = mapped_column(String, nullable=True)
    location: Mapped[str | None] = mapped_column(String, nullable=True)
